design/RELEASE_CADENCE.md

```diff
@@ -0,0 +1,67 @@
+# MetalLB release cadence
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+- [Proposal](#proposal)
+  - [Notes/Constraints/Caveats (Optional)](#notesconstraintscaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (Optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Summary
+
+The purpouse of this enhancement is to define a versioning and release schedule tied to
+kubernetes versions.
+
+## Motivation
+
+The reason for this proposal is to establish a dependency between a given MetalLB version and kubernetes versions.
+
+This will make MetalLB more reliable with regards with the set of kubernetes apis it can use or can't use. A good example for it is the introduction of the use of EndpointSlices, which are versioned as beta in k8s 1.17+ but moving to stable in 1.21.
+
+### Goals
+
+Define MetalLB release cadence and versioning as bounded to kubernetes versions.
+
+## Proposal
+
+Each MetalLB release branch correspond to a released k8s version.
```

Comment:
Sorry for the silly question but from here I understand that we will have a 1-1 mapping between kubernetes minor releases and MetalLB. For example, k8s 1.20.x <---> MetalLB X. I guess if 1.21 is released, we will release MEtalLB X+1 from that part too.

However, later the proposal says:
> The most recent release of MetalLB must be able to support all the supported k8s versions (n-3 as of today). We can ensure that by having multiple CI lanes running kind
with different kubernetes versions.

Why would say "MetalLB X works with k8s 1.20" and have a 1-1 mapping, when it also works with n-3 releases?

Reply:
Maybe the reason is: https://github.com/metallb/metallb/pull/858#discussion_r700287211 ? So, not "works with 1.20", but just having the last release "up to date"?